app/dashboard/components/projectFilter.tsx

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect, useMemo } from 'react';
+
+//reflect uml class once definitive
+interface Project {
+	id: number;
+	title: string;
+	date: string;
+}
+
+interface FilterProjectsProps {
+	projects: Project[];
+}
+
+interface FilteredProjectsResult {
+	loading: boolean;
+	filter: string;
+	filteredProjects: Project[];
+	handleFilterChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export default function FilterProjects({ projects }: FilterProjectsProps): FilteredProjectsResult {
+	const [filter, setFilter] = useState<string>('');
+	const [loading, setLoading] = useState<boolean>(true); //new loading state
+
+	const timeout = setTimeout(() => {
```

Comment:
Why?

Reply:
De loading gaat gebruikt worden om op de view een skeleton-page weer te geven tijdens het laden van de data, hiervoor is deze state nodig.